lib/ansible/plugins/connection/kubectl.py

```diff
@@ -23,47 +23,127 @@
 DOCUMENTATION = """
     author:
         - xuxinkun
+    
     connection: kubectl
-    short_description: Run tasks in kubernetes pods
+    
+    short_description: Execute tasks in pods running on Kubernetes. 
+    
     description:
-        - Run commands or put/fetch files to an existing kubernetes pod container.
-    version_added: "2.0"
+        - Use the kubectl exec command to run tasks in, or put/fetch files to, pods running on the Kubernetes
+          container platform.
+    
+    version_added: "2.5"
+    
+    requirements:
+      - kubectl (go binary)
+    
     options:
-      kube_namespace:
+      kubectl_container:
         description:
-            - The namespace of the pod
+          - Container name. Required when a pod contains more than one container.
         default: ''
         vars:
-            - name: ansible_kube_namespace
-            - name: ansible_kubectl_namespace
+          - name: ansible_kubectl_container
+      kubectl_namespace:
+        description:
+          - The namespace of the pod
+        default: ''
+        vars:
+          - name: ansible_kubectl_namespace
+        env:
+          - name: K8S_AUTH_NAMESPACE
       kubectl_extra_args:
         description:
-            - Extra arguments to pass to the kubectl command line
+          - Extra arguments to pass to the kubectl command line.
+        default: ''
+        vars:
+          - name: ansible_kubectl_extra_args
+        env:
+          - name: K8S_AUTH_EXTRA_ARGS
+      kubectl_kubeconfig:
+        description:
+          - Path to a kubectl config file. Defaults to I(~/.kube/conig)
+        default: ''
+        vars:
+          - name: ansible_kubectl_kubeconfig
+          - name: ansible_kubectl_config
+        env:
+          - name: K8S_AUTH_KUBECONFIG
+      kubectl_context:
+        description:
+          - The name of a context found in the K8s config file.
+        default: ''
+        vars:
+          - name: ansible_kubectl_context
+        env:
+          - name: k8S_AUTH_CONTEXT
+      kubectl_host:
```

Comment:
I prefer to use `server` instead of `host`. Because `host` in ansible point to the target of operation. BTW, kubectl also use `-s, --server string                                The address and port of the Kubernetes API server`

Reply:
I know. There's an alias for 'server'. Calling it `host` keeps it consistent with the modules and Python client. As a user, you can use `server` or `host`. The Python client calls it `host`, and uses `HOST` in the env variable.

Reply:
How about adding a SERVER env variable?

Reply:
Done.